redhat-appstudio/service-provider-integration-operator
# Manual token upload for Quay

### What does this PR do?
Allows manual username + password (typically robot account) upload for Quay

### Screenshot/screencast of this PR

![image](https://user-images.githubusercontent.com/1651062/164288135-2c42db05-5991-4fc5-a565-21bc0fa89ee3.png)
![image](https://user-images.githubusercontent.com/1651062/164288168-b00290c8-a523-48f4-abf6-ffdd07608a0b.png)


### What issues does this PR fix or reference?

https://issues.redhat.com/projects/SVPI/issues/SVPI-107

### How to test this PR?

 - Deploy SPI from images `mshaposh/spio:latest` and `mshaposh/spis:latest`.  
 - Prepare robot account cred--s on Quay
 - Create AccessTokenBinding as usual with quay.io URL
 - Grab the autogenerated AccessToken name
  - Call POST `https://<host>/token/<namespace>/<access_token_name>` with K8S authentication and
  POST data like
  ```
  {
  "username": "quay+user_demo",
  "access_token": "TPQ8OQSY8CC7SNLKH7",
  }
 ```
  - Make sure AT is provisioned as well as secrets created in user namespace etc
